parlai/core/torch_classifier_agent.py

```diff
@@ -168,6 +171,133 @@ def value(self) -> float:
             return numer / denom
 
 
+class AUCMetrics(Metric):
+    """
+    Computes Area Under ROC Curve (AUC) metrics.
+
+    Does so by keeping track of positives' and negatives' probability score counts in
+    Counters or dictionaries. Note the introduction of `max_bucket_dec_places`; this
+    integer number determines the number of digits to save for the probability scores. A
+    higher `max_bucket_dec_places` will a more accurate estimate of the exact AUC
+    metric, but may also use more space.
+    """
+
+    @property
+    def macro_average(self) -> bool:
+        """
+        Indicates whether this metric should be macro-averaged when globally reported.
+        """
+        return False
+
+    @classmethod
+    def raw_data_to_auc(
+        cls,
+        true_labels: List[Union[int, str]],
+        pos_probs: List[float],
+        class_name,
+        max_bucket_dec_places: int = 3,
+    ):
+        auc_object = cls(class_name, max_bucket_dec_places=max_bucket_dec_places)
+        auc_object.update_raw(
+            true_labels=true_labels, pos_probs=pos_probs, class_name=class_name
+        )
+        return auc_object
+
+    def __init__(
+        self,
+        class_name: Union[int, str],
+        pos_dict: Count[float] = None,
+        neg_dict: Count[float] = None,
+        max_bucket_dec_places: int = 3,
+    ):
+        self._pos_dict = pos_dict if pos_dict else Counter()
+        self._neg_dict = neg_dict if neg_dict else Counter()
+        self._class_name = class_name
+        self._max_bucket_dec_places = max_bucket_dec_places
+
+    def update_raw(
+        self, true_labels: List[Union[int, str]], pos_probs: List[float], class_name
+    ):
+        assert self._class_name == class_name
```

Comment:
Currently, we only have 2 classes, but if there were more classes and we use this for a one-vs rest approach, it would be a security measure to make sure that the classes won't get mixed up for some reason; more comments sounds like a good idea :)